apps/notifications/src/chain-events/repositories/block-cursor/block-cursor.repository.ts

```diff
@@ -0,0 +1,63 @@
+import { InjectDrizzle } from '@knaadh/nestjs-drizzle-pg';
+import { Injectable } from '@nestjs/common';
+import { eq } from 'drizzle-orm';
+import { NodePgDatabase } from 'drizzle-orm/node-postgres';
+
+import { DRIZZLE_PROVIDER_TOKEN } from '@src/config/db.config';
+import { valueBackoff } from '@src/lib/value-backoff/value-backoff';
+import * as schema from '../../model-schemas';
+
+@Injectable()
+export class BlockCursorRepository {
+  constructor(
+    @InjectDrizzle(DRIZZLE_PROVIDER_TOKEN)
+    private readonly db: NodePgDatabase<typeof schema>,
+  ) {}
+
+  private readonly id = 'latest';
+
+  async getNextBlockForProcessing<T>(
+    cb: (block: number) => Promise<T>,
+  ): Promise<T> {
+    return await this.db.transaction(async (transaction) => {
```

Comment:
question(blocking): do we want to do a retry inside a db transaction? I think we will end up in the same situation as the start trial

Reply:
Ah ok makes sense, but then if we spin up other workers we could potentially stack db connections if they all try to access the blocks at the same time, but I don't think we'll need more workers any time soon so all good.

Reply:
In case we need more workers, we could have more ids than just "latest" and adjust the query. But yea, I don't see a need for this